phoenix-core/src/main/java/org/apache/phoenix/monitoring/connqueryservice/ConnectionQueryServicesMetricsManager.java

```
@@ -0,0 +1,375 @@
+     * thread safe manner)
+     * @return returns instance of ConnectionQueryServicesMetricsManager for the said
+     *         ConnectionQueryService
+     */
+    public static ConnectionQueryServicesMetricsManager getInstance() {
+        ConnectionQueryServicesMetricsManager localRef = connectionQueryServicesMetricsManager;
+
+        if (localRef == null) {
+            synchronized (ConnectionQueryServicesMetricsManager.class) {
+                localRef = connectionQueryServicesMetricsManager;
+                if (localRef == null) {
+                    QueryServicesOptions options = QueryServicesOptions.withDefaults();
+                    localRef = connectionQueryServicesMetricsManager =
+                            new ConnectionQueryServicesMetricsManager(options);
+                    LOGGER.info("connection query service created object for metrics manager");
+                    if (isConnectionQueryServiceMetricsEnabled
+                            && isConnectionQueryServiceMetricPublisherEnabled) {
+                        String className = options.getConnectionProfileMetricsPublisherClass();
+                        if (className != null) {
+                            MetricServiceResolver mResolver = new MetricServiceResolver();
+                            LOGGER.info(String.format(
+                                    "connection query service metrics publisher className %s",
+                                    className));
+                            try {
+                                mPublisher = mResolver.instantiate(className);
+                                mPublisher.registerMetricProvider();
+                            } catch (Throwable e) {
+                                LOGGER.error("The exception from metric publish Function", e);
+                            }
+
+                        } else {
+                            LOGGER.warn("connection query service metrics publisher className"
```

is it better to throw an exception if this is not acceptable condition?

---

It's better to not kill a service/JVM start for Metrics classname issue.